src/tags.psm1

```diff
@@ -0,0 +1,1374 @@
+If an empty array is passed in, then all tags will be removed from the asset.
+
+.INPUTS
+None.
+
+.OUTPUTS
+JSON response from Safeguard Web API.
+
+.EXAMPLE
+Update-SafeguardAssetTag assetname -Tag @("Prod,VM,DMZ")
+
+.EXAMPLE
+Update-SafeguardAssetTag assetname -Tag @(1,2,3)
+
+.EXAMPLE
+Update-SafeguardAssetTag 8 -Tag @("Prod,VM,DMZ")
+
+.EXAMPLE
+$tags = @("VM","Prod","DMZ")
+Update-SafeguardAssetTags -Asset "assetName" -Tag $tags
+
+#>
+function Update-SafeguardAssetTag {
+    [CmdletBinding()]
+    Param(
+        [Parameter(Mandatory=$false)]
+        [string]$Appliance,
+        [Parameter(Mandatory=$false)]
+        [object]$AccessToken,
+        [Parameter(Mandatory=$false)]
+        [switch]$Insecure,
+        [Parameter(Mandatory=$false)]
+        [object]$AssetPartition,
+        [Parameter(Mandatory=$false)]
+        [int]$AssetPartitionId = $null,
+        [Parameter(Mandatory=$true,Position=0)]
+        [object]$Asset,
+        [Parameter(Mandatory=$false)]
+        [object[]]$Tag
+    )
+
+    if (-not $PSBoundParameters.ContainsKey("ErrorAction")) { $ErrorActionPreference = "Stop" }
+    if (-not $PSBoundParameters.ContainsKey("Verbose")) { $VerbosePreference = $PSCmdlet.GetVariableValue("VerbosePreference") }
+   
+    # we need to use function Resolve-SafeguardAssetId from the asset.psm1 module to get the Asset ID so check if the module is load
+    if (-not (Get-Module assets)) {
+        Import-module "$PSScriptRoot\assets.psm1"
+    }
+    $AssetId = (Resolve-SafeguardAssetId -Appliance $Appliance -AccessToken $AccessToken -Insecure:$Insecure -AssetPartition $AssetPartition -AssetPartitionId $AssetPartitionId -Asset $Asset)    
+    $local:body =@() # empty array to store the tag IDs.
+    if ($Tag.Count -gt 0) {
+        foreach ($tagObj in $Tag) {
+            $tagId = Resolve-SafeguardTagId $tagObj
+            if ($tagId) {
+                # found the tag so add the ID to the body
+                $local:body += [PSCustomObject]@{ Id = $tagId }
+            }
+        }
+        Invoke-SafeguardMethod -AccessToken $AccessToken -Appliance $Appliance -Insecure:$Insecure Core PUT "Assets/$AssetId/Tags" -Body $local:body
+    } else {
+        # No tags specified so remove all tags. Send empty json body.
+        Invoke-SafeguardMethod -AccessToken $AccessToken -Appliance $Appliance -Insecure:$Insecure Core PUT "Assets/$AssetId/Tags" -JsonBody "[]"
+    }
+}
+
+
+<#
+.SYNOPSIS
+Get the tags from a specific account via the Web API.
+
+.DESCRIPTION
+Get the assigned tags for a specific account.
+
+.PARAMETER Appliance
+IP address or hostname of a Safeguard appliance.
+
+.PARAMETER AccessToken
+A string containing the bearer token to be used with Safeguard Web API.
+
+.PARAMETER Insecure
+Ignore verification of Safeguard appliance SSL certificate.
+
+.PARAMETER AssetPartition
+An integer containing an ID or a string containing the name of the asset partition to get tags from an asset.
```

Comment:
"...to get tags from an asset" should be "...to get tags from an asset account", or "...to get tags from an account."